bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -239,57 +250,80 @@ private void checkConnection() throws CoolMasterClientError {
                     throw new IOException("Unexpected data following prompt");
                 }
                 logger.trace("Buffer empty following solicited > prompt");
-            } catch (IOException e) {
+            } catch (final IOException ioe) {
                 disconnect();
-                logger.debug("{}", e.getLocalizedMessage(), e);
-                throw new CoolMasterClientError(
-                        String.format("No response from CoolMasterNet unit %s:%s", config.host, config.port));
+                throw ioe;
             }
         }
     }
 
+    /**
+     * Opens the socket.
+     *
+     * <p>
+     * Guarantees to either open the socket or thrown an exception.
+     *
+     * @throws IOException if the socket could not be opened
+     */
     private void connect() throws IOException {
         synchronized (socketLock) {
-            ControllerConfiguration config = getConfigAs(ControllerConfiguration.class);
+            final ControllerConfiguration cfg = getConfigAs(ControllerConfiguration.class);
             try {
-                Socket localSocket = new Socket();
-                localSocket.connect(new InetSocketAddress(config.host, config.port), SOCKET_TIMEOUT);
-                localSocket.setSoTimeout(SOCKET_TIMEOUT);
-                socket = localSocket;
-            } catch (UnknownHostException e) {
-                logger.error("Unknown socket host: {}", config.host);
-                socket = null;
-            } catch (SocketException e) {
-                logger.error("Failed to connect to {}:{}: {}", config.host, config.port, e.getLocalizedMessage(), e);
+                logger.debug("Connecting to {}:{}", cfg.host, cfg.port);
+                final Socket socket = new Socket();
+                socket.connect(new InetSocketAddress(cfg.host, cfg.port), SOCKET_TIMEOUT_MS);
+                socket.setSoTimeout(SOCKET_TIMEOUT_MS);
+                this.socket = socket;
+            } catch (final IOException ioe) {
                 socket = null;
+                throw ioe;
             }
         }
     }
 
+    /**
+     * Attempts to disconnect the socket.
+     *
+     * <p>
+     * Disconnection failure is not considered an error, although will be logged.
+     */
     private void disconnect() {
         synchronized (socketLock) {
-            Socket localSocket = socket;
-            if (localSocket != null) {
+            final ControllerConfiguration cfg = getConfigAs(ControllerConfiguration.class);
```

Comment:
It would be more optimal to save the ControllerConfiguration to a field during initialization so you don't need to create it each time. Creating the configuration class is a costly procedure.